### PR TITLE
Bumped versions of dependencies to resolve 40+ vulnerabilities for v3.x.x

### PR DESCRIPTION
**Pull request by NSpencer4**
### All Submissions:

- [X] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [X] Have you written unit tests?
- [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [X] This PR is associated with an existing issue?

**Closing issues**

Put `closes #XXXX` (where XXXX is the issue number) in your comment to auto-close the issue that your PR fixes.

### If this is a new feature submission:

- [ ] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Test plan**
This was a dependency change as well as minor type linting issues I fixed. I ran the unit tests and they all passed after the dependencies were updated.

### CODE REVIEW

**Comment by NSpencer4** [comment]
closes #1158

File: src/metadataGeneration/typeResolver.ts [hunk 1]
```diff
@@ -657,7 +657,7 @@ export class TypeResolver {
 
         const moduleDeclaration = node as ts.ModuleDeclaration | ts.EnumDeclaration;
         return (moduleDeclaration.name as ts.Identifier).text.toLowerCase() === leftmostName.toLowerCase();
-      }) as Array<ts.ModuleDeclaration | ts.EnumDeclaration>;
+      }) as [ts.ModuleDeclaration | ts.EnumDeclaration];
```
Comment (tsimbalar) [inline src/metadataGeneration/typeResolver.ts:660]: are those types equivalent ? 

wouldn't it be : 

```suggestion
      }) as (ts.ModuleDeclaration | ts.EnumDeclaration)[];
```

?

File: src/metadataGeneration/typeResolver.ts [hunk 2]
```diff
@@ -758,7 +758,7 @@ export class TypeResolver {
       .filter(member => !isIgnored(member))
       .filter(member => member.kind === ts.SyntaxKind.PropertyDeclaration)
       .filter(member => !this.hasStaticModifier(member))
-      .filter(member => this.hasPublicModifier(member)) as Array<ts.PropertyDeclaration | ts.ParameterDeclaration>;
+      .filter(member => this.hasPublicModifier(member)) as [ts.PropertyDeclaration | ts.ParameterDeclaration];
```
Comment (tsimbalar) [inline src/metadataGeneration/typeResolver.ts:761]:
```suggestion
      .filter(member => this.hasPublicModifier(member)) as (ts.PropertyDeclaration | ts.ParameterDeclaration)[];
```

File: src/metadataGeneration/typeResolver.ts (hunk 1 above)
Comment (NSpencer4) [reply src/metadataGeneration/typeResolver.ts]: Yes! Good catch. Updated!

File: src/metadataGeneration/typeResolver.ts (hunk 2 above)
Comment (NSpencer4) [reply src/metadataGeneration/typeResolver.ts]: Yes! Good catch. Updated!

**Comment by WoH** [comment]
See https://github.com/lukeautry/tsoa/issues/1158. If you want to update the 3.x branch, merge the changes up to a9cf0aaad0f88acbde1d571a8481dba34830538c first. Then these "security vulnerabilities" will also mostly disappear.

**Pull request closed without merging**